src/runtime_src/core/tools/common/XBUtilities.cpp

```diff
@@ -159,4 +162,229 @@ XBUtilities::trace_print_tree(const std::string & _name,
+
+        // See if we have a short flag
+        if (optionDisplayName.size() == SHORT_OPTION_STRING_SIZE)
+          continue;
+
+        // We are not interested in any arguments
+        if (option->semantic()->max_tokens() > 0)
+          continue;
+
+        // This option shouldn't be required
+        if (option->semantic()->is_required() == true) 
+          continue;
+
+        std::string completeOptionName = option->canonical_display_name(po::command_line_style::allow_long);
+        buffer << " [" << completeOptionName << "]";
+      }
+    }
+
+    // Gather up the options with arguments
+    for (auto option : options) {
+      // Skip if there are no arguments
+      if (option->semantic()->max_tokens() == 0)
+        continue;
+
+      // This option shouldn't be required
+      if (option->semantic()->is_required() == true) 
+        continue;
+
+      
+      std::string completeOptionName = option->canonical_display_name(po::command_line_style::allow_dash_for_short);
+      buffer << " [" << completeOptionName << " arg]";
+    }
+
+    // Gather up the required options with arguments
+    for (auto option : options) {
+      // Skip if there are no arguments
+      if (option->semantic()->max_tokens() == 0)
+        continue;
+
+      // This option is required
+      if (option->semantic()->is_required() == false) 
+        continue;
+
+      std::string completeOptionName = option->canonical_display_name(po::command_line_style::allow_dash_for_short);
+      buffer << " " << completeOptionName << " arg";
+    }
+
+  return buffer.str();
+}
+
+void 
+XBUtilities::wrap_paragraph( const std::string _unformattedString, 
+                             unsigned int _indentWidth, 
+                             unsigned int _columnWidth, 
+                             bool _indentFirstLine,
+                             std::string &_formattedString)
+{
+  // Set return variables to a now state
+  _formattedString.clear();
+
+  if (_indentWidth >= _columnWidth) {
+    std::string errMsg = boost::str(boost::format("Internal Error: %s paragraph indent (%d) is greater than or equal to the column width (%d) ") % __FUNCTION__ % _indentWidth % _columnWidth);
+    throw std::runtime_error(errMsg);
+  }
+
+  const unsigned int paragraphWidth = _columnWidth - _indentWidth;
+
+  std::string::const_iterator lineBeginIter = _unformattedString.begin();
+  const std::string::const_iterator paragraphEndIter = _unformattedString.end();
+
+  unsigned int linesProcessed = 0;
+
+  while (lineBeginIter < paragraphEndIter)  
+  {
+    // Remove leading spaces
+    if ((linesProcessed > 0) && 
+        (*lineBeginIter == ' ')) {
+      lineBeginIter++;
+      continue;
+    }
+
+    // Determine the end-of-the line to be examined
+    std::string::const_iterator lineEndIter = lineBeginIter;
+    int64_t remainingChars = std::distance(lineBeginIter, paragraphEndIter);

```

Comment:
int64_t?

Reply:
std::distance returns an uint64_t value, and I had assigned it to an unsigned int value originally.  While this worked nicely on Linux it failed on Windows, hence the uint64_t.

Since this is causing some confusion, I instead replaced it with an auto which works both on windows and linux.